### PR TITLE
Initial file setup

### LINKED ISSUE #9: Initialize repository with create react app --typescript

### LINKED ISSUE #10: Set up file structure
* Components in individual folders (js file, test file, css file)
* Api calls in isolated locations

### PR DESCRIPTION
**Pull request by timkeresey**
### What does this PR do?

Sets up the files structure for all components and installs testing dependencies

### Where should the reviewer start?

the src directory of the project.

### How should this be manually tested?

Through typescript's built-in debugger.

### Any background context you want to provide?

None

### What are the relevant tickets?

- closes #10, closes #9,   

### Questions:

None

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -3,15 +3,18 @@
   "version": "0.1.0",
   "private": true,
   "dependencies": {
-    "@testing-library/jest-dom": "^5.11.4",
-    "@testing-library/react": "^11.1.0",
-    "@testing-library/user-event": "^12.1.10",
+    "@testing-library/jest-dom": "^5.11.8",
```
Comment (priyapower) [inline package.json:6]: I see we are using jest for our testing environment in FE

**Review by BlakeDonn — APPROVED** [review]
Structure looks good, I think the yarn.lock is redundant as there is already a package.lock. I will research this further to see if we can remove it

File: package.json [hunk 2]
```diff
@@ -3,15 +3,18 @@
...
+    "@testing-library/react": "^11.2.2",
+    "@testing-library/user-event": "^12.6.0",
     "@types/jest": "^26.0.15",
     "@types/node": "^12.0.0",
     "@types/react": "^16.9.53",
     "@types/react-dom": "^16.9.8",
+    "prop-types": "^15.7.2",
     "react": "^17.0.1",
```
Comment (Atos20) [inline package.json:14]: Thanks for the dependencies, at least we have our testing environment set up ready to go.

File: package.json [hunk 3]
```diff
@@ -3,15 +3,18 @@
...
     "react-dom": "^17.0.1",
+    "react-player": "^2.7.2",
```
Comment (Atos20) [inline package.json:16]: probably that was me but I don't think we will need `react-player`, make sure we uninstall it before deplying

File: src/components/App/App.tsx [hunk 4]
```diff
@@ -0,0 +1,38 @@
+import React, { useState, useEffect } from 'react';
+
+//types
+import { data } from '../../types';
```
Comment (Atos20) [inline src/components/App/App.tsx:4]: Thanks for the initial set up my friend.

**Review by Atos20 — CHANGES_REQUESTED** [review]
Tim, I noticed you didn't do an `interactive rebase`, could you please do it before merging this PR. 
I already forgot how to do it but here is a helpful [link](https://medium.com/@praveenmuth2/interactive-rebase-the-ultimate-git-superpower-d43b94afbf01#8090) to help you do it. Let me know if you need help.


Thanks

File: src/components/App/App.tsx [hunk 5]
```diff
@@ -0,0 +1,38 @@
+import React, { useState, useEffect } from 'react';
```
Comment (priyapower) [inline src/components/App/App.tsx:1]: To me this looks like a "runner" file for our app

File: src/components/Error/Error.tsx [hunk 6]
```diff
@@ -0,0 +1,11 @@
+import React, { useState, useEffect } from 'react';
```
Comment (priyapower) [inline src/components/Error/Error.tsx:1]: I believe this is a basic template for your error handling

File: src/react-app-env.d.ts [hunk 7]
```diff
@@ -1 +1,6 @@
 /// <reference types="react-scripts" />
+
```
Comment (priyapower) [inline src/react-app-env.d.ts:2]: I am unsure what this does, any suggestions for resources/googling?

File: src/types.tsx [hunk 8]
```diff
@@ -0,0 +1,4 @@
+export interface data {
```
Comment (priyapower) [inline src/types.tsx:1]: This looks like a template for exporting your data
Does this data come from your store?

**Review by priyapower — APPROVED** [review]
Looks great! I had some questions, please respond at your convenience

**Review by Atos20 — APPROVED** [review]

File: src/react-app-env.d.ts (hunk 7 above)
Comment (timkeresey) [reply src/react-app-env.d.ts:2]: I'm unsure too. I added this after receiving an error and got from this resource: https://github.com/facebook/create-react-app/issues/10109

File: src/types.tsx (hunk 8 above)
Comment (timkeresey) [reply src/types.tsx:1]: This is an interface/type declaration for typescript. This is just a placeholder until we define it.

File: src/react-app-env.d.ts (hunk 7 above)
Comment (priyapower) [reply src/react-app-env.d.ts:2]: Cool, thank you!

File: src/types.tsx (hunk 8 above)
Comment (priyapower) [reply src/types.tsx:1]: Very interesting! Thanks for the information!

**Pull request merged**